subprojects/docs/src/docs/userguide/commandLineInterface.adoc

```diff
@@ -0,0 +1,520 @@
+
+==== Listing tasks
+
+Running `gradle tasks` gives you a list of the main tasks of the selected project. This report shows the default tasks for the project, if any, and a description for each task.
+
+----
+❯ gradle tasks
+----
+
+By default, this report shows only those tasks which have been assigned to a task group. You can obtain more information in the task listing using the `--all` option.
+
+----
+❯ gradle tasks --all
+----
+
+[[sec:show_task_details]]
+==== Show task usage details
+
+Running `gradle help --task someTask` gives you detailed information about a specific task.
+
+++++
+<sample id="taskHelp" dir="userguide/tutorial/projectReports" title="Obtaining detailed help for tasks">
+    <output args="-q help --task libs"/>
+</sample>
+++++
+
+This information includes the full task path, the task type, possible command line options and the description of the given task.
+
+==== Reporting dependencies
+
+Build scans give a full, visual report of what project and binary dependencies exist on which configurations, transitive dependencies, and dependency version selection.
+
+----
+❯ gradle myTask --scan
+----
+
+This will give you a link to a web-based report, where you can find dependency information like this.
+
+image::img/gradle-core-test-build-scan-dependencies.png[Build Scan dependencies report]
+
+==== Listing project dependencies
+
+Running `gradle dependencies` gives you a list of the dependencies of the selected project, broken down by configuration. For each configuration, the direct and transitive dependencies of that configuration are shown in a tree. Below is an example of this report:
+
+----
+❯ gradle dependencies
+----
+
+Running `gradle buildEnvironment` visualises the buildscript dependencies of the selected project, similarly to how `gradle dependencies` visualizes the dependencies of the software being built.
+
+----
+❯ gradle buildEnvironment
+----
+
+Running `gradle dependencyInsight` gives you an insight into a particular dependency (or dependencies) that match specified input.
+
+----
+❯ gradle dependencyInsight
+----
+
+Since a dependency report can get large, it can be useful to restrict the report to a particular configuration. This is achieved with the optional `--configuration` parameter:
+
+++++
+<sample id="dependencyListReportFiltered" dir="userguide/tutorial/projectReports" title="Filtering dependency report by configuration">
+    <output args="-q api:dependencies --configuration testCompile"/>
+</sample>
+++++
+
+[[sec:listing_properties]]
+==== Listing project properties
+
+Running `gradle properties` gives you a list of the properties of the selected project.
+
+++++
+<sample id="propertyListReport" dir="userguide/tutorial/projectReports" title="Information about properties">
+    <output args="-q api:properties" ignoreExtraLines="true"/>
+</sample>
+++++
+
+==== Software Model reports
```

Comment:
Let's remove this section. Software model documentation should already have been removed...

Reply:
I'd like to, but I'd feel pretty guilty about it. I imagine there are lots of existing projects using Gradle 4.5+ that still use the software model in the native space. 

I'd feel much better about it if we _really_ deprecated this feature with warnings when one tries to use a `model {}` block or something. As it stands this feature only feels partially deprecated (perhaps for good reason). 

Perhaps @lacasseio or @big-guy would help us decide on this?

Reply:
+1 for leaving it in until we have a production-ready version of the new plugins or officially deprecated this feature.